app/GraphQL/Ecosystem/Mutations/Roles/RolesManagementMutation.php

```diff
@@ -70,6 +72,28 @@ public function removeRoleFromUser(mixed $rootValue, array $request): bool
         return true;
     }
 
+    public function givePermissionToRole(mixed $rootValue, array $request): bool
+    {
+        $systemModule = SystemModulesRepository::getByModelName($request['systemModule'], app(Apps::class));
+        Bouncer::allow($request['role'])->to($request['permission'], $systemModule->model_name);
+
+        $roles = RolesRepository::getMapAbilityInModules($request['role']);
+        Redis::set('roles:abilities', $roles);
+
+        return true;
+    }
+
+    public function removePermissionToRole(mixed $rootValue, array $request): bool
+    {
+        $systemModule = SystemModulesRepository::getByModelName($request['systemModule'], app(Apps::class));
+        Bouncer::disallow($request['role'])->to($request['permission'], $systemModule->model_name);
+
+        $roles = RolesRepository::getMapAbilityInModules($request['role']);
+        Redis::set('roles:abilities', $roles);
```

Comment:
@FredPeal i would move the roles:abilities to a enum so I don't have to worry about that string in 2 diff places